frontend/src/routes/[[lang=locale]]/candidate/(protected)/+page.svelte

```diff
@@ -6,22 +6,141 @@
 
-  const {userStore} = getContext<CandidateContext>('candidate');
+  const {
+    userStore,
+    basicInfoFilledStore,
+    nofUnasweredBasicInfoQuestionsStore: nofUnansweredBasicInfoQuestions,
+    opinionQuestionsFilledStore,
+    nofUnansweredOpinionQuestionsStore: nofUnansweredOpinionQuestions,
+    questionsStore
+  } = getContext<CandidateContext>('candidate');
   const user = get(userStore);
   const userName = user?.candidate?.firstName;
+
+  let dataEditable: boolean;
+
+  let questions = get(questionsStore) ?? [];
+
+  if (questions) {
+    //TODO: use store when store is implemented
+    dataEditable = Object.values(questions)[0].editable;
+  }
+
+  let opinionQuestionsLeft: number | undefined;
+  nofUnansweredOpinionQuestions?.subscribe((value) => {
+    opinionQuestionsLeft = value;
+  });
+
+  let opinionQuestionsFilled: boolean | undefined;
+  opinionQuestionsFilledStore?.subscribe((value) => {
+    opinionQuestionsFilled = value;
+  });
+
+  let basicInfoQuestionsLeft: number | undefined;
+  nofUnansweredBasicInfoQuestions?.subscribe((value) => {
+    basicInfoQuestionsLeft = value;
+  });
+
+  let basicInfoFilled: boolean | undefined;
+  basicInfoFilledStore?.subscribe((value) => {
+    basicInfoFilled = value;
+  });
+
+  function getNextAction() {
+    if (!!basicInfoFilled && !!opinionQuestionsFilled) {
+      return {
+        title: $t('candidateApp.homePage.title'),
+        explanation: $t('candidateApp.homePage.description'),
+        tip: $t('candidateApp.homePage.tip'),
+        buttonTextBasicInfo: $t('candidateApp.homePage.basicInfoButtonView'),
+        buttonTextQuestion: $t('candidateApp.homePage.questionsButtonView'),
+        buttonTextPrimaryActions: $t('candidateApp.homePage.previewButton'),
+        href: $getRoute(Route.CandAppPreview)
+      };
+    } else if (!!basicInfoFilled && !opinionQuestionsFilled) {
+      return {
+        title: $t('candidateApp.homePage.greeting', {userName}),
+        explanation: $t('candidateApp.homePage.explanation'),
+        buttonTextBasicInfo: $t('candidateApp.homePage.basicInfoButtonEdit'),
+        buttonTextQuestion: $t('candidateApp.homePage.questionsButton'),
+        buttonTextPrimaryActions: $t('candidateApp.homePage.questionsButton'),
+        href: $getRoute(Route.CandAppQuestions)
+      };
+    } else {
+      return {
+        title: $t('candidateApp.homePage.greeting', {userName}),
+        explanation: $t('candidateApp.homePage.explanation'),
+        buttonTextBasicInfo: $t('candidateApp.homePage.basicInfoButton'),
+        buttonTextQuestion: $t('candidateApp.homePage.questionsButton'),
+        buttonTextPrimaryActions: $t('candidateApp.homePage.basicInfoButtonPrimaryActions'),
+        href: $getRoute(Route.CandAppProfile)
+      };
+    }
+  }
 </script>
 
 <!--Homepage for the user-->
 
-<BasicPage title={$t('candidateApp.homePage.greeting', {userName})}>
+<BasicPage title={getNextAction().title}>
```

Comment:
The `getNextAction()` function is invoked quite many times in the below code. It would be much more efficient to store the value in a variable and refer to that.

Reply:
I think this change should still be made. Would something like this work?
```
<script>
    ...
    $: nextAction = getNextAction();
</script

<BasicPage title={nextAction.title}>
    ...
</BasicPage>
```